doc/NAT/SAI-NAT-API.md

```diff
@@ -0,0 +1,443 @@
+nat_entry_attr[4].value.u32 = 100;
+
+nat_entry_attr[5].id = SAI_NAT_ENTRY_ATTR_ENABLE_PACKET_COUNT;
+nat_entry_attr[5].value.bool = true;
+
+nat_entry_attr[6].id = SAI_NAT_ENTRY_ATTR_ENABLE_BYTE_COUNT;
+nat_entry_attr[6].value.bool = true;
+
+attr_count = 7;
+
+memset(&snat_etnry, 0, sizeof(nat_etnry));
+
+snat_entry.data.key.src_ip = 10.0.0.1;
+snat_entry.data.mask.src_ip = 0xffffffff;
+snat_entry.data.key.l4_src_port = 6000;
+snat_entry.data.mask.l4_src_port = 0xffff;
+snat_entry.data.key.proto = 17;
+snat_entry.data.mask.proto = 0xff;
+
+
+create_nat_entry(&snat_entry, attr_count, nat_entry_attr);
+
+```
+
+##### Step 2: Create a Destination NAT Entry object:
+```sh
+
+sai_attribute_t nat_entry_attr[10];
+nat_entry_t dnat_entry;
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_NAT_TYPE;
+nat_entry_attr[0].value = SAI_NAT_TYPE_DESTINATION_NAT;
+
+nat_entry_attr[1].id = SAI_NAT_ENTRY_ATTR_DST_IP;
+nat_entry_attr[1].value.u32 = 10.0.0.1; //example string
+
+nat_entry_attr[2].id = SAI_NAT_ENTRY_ATTR_L4_DST_PORT;
+nat_entry_attr[2].value.u16 = 6000;
+
+nat_entry_attr[3].id = SAI_NAT_ENTRY_ATTR_TO_ZONE;
+nat_entry_attr[3].value.u32 = 100;
+
+nat_entry_attr[4].id = SAI_NAT_ENTRY_ATTR_FROM_ZONE;
+nat_entry_attr[4].value.u32 =200;
+
+nat_entry_attr[5].id = SAI_NAT_ENTRY_ATTR_ENABLE_PACKET_COUNT;
+nat_entry_attr[5].value.bool = true;
+
+nat_entry_attr[6].id = SAI_NAT_ENTRY_ATTR_ENABLE_BYTE_COUNT;
+nat_entry_attr[6].value.bool = true;
+
+attr_count = 7;
+
+memset(&dnat_etnry, 0, sizeof(nat_etnry));
+
+dnat_entry.data.key.dst_ip = 65.55.42.1;
+dnat_entry.data.mask.dst_ip = 0xffffffff;
+dnat_entry.data.key.l4_dst_port = 1024;
+dnat_entry.data.mask.l4_dst_port = 0xffff;
+dnat_entry.data.key.proto = 17;
+dnat_entry.data.mask.proto = 0xff;
+
+create_nat_entry(&dnat_entry, attr_count, nat_entry_attr);
+
+```
+
+### 5.2 Double NAT
+Double NAT is a nat variant where both the Src and Dest IP gets modified when a packet crosses the address zone. Generally used when conflicting addresses are used in the private networks which conflicts with the public IP addresses. Double NAT overrides SNAT or DNAT actions in the SAI pipeline. Reverse entry should be created in similar manner.
+
+> From Zone ID: 100
+To Zone ID: 200
+VRF: None
+Packet Count: Enable
+Byte Count: Enable
+SNAT Entry
+ExternalEndpoint: 138.76.28.1 
+InternalEndpoint: 200.200.200.1
+DNAT Entry
+ExternalEndpoint: 200.200.200.100
+InternalEndpoint: 172.16.1.100
+
+##### Step 1: Create a Double NAT Entry:
+```sh
+sai_attribute_t nat_entry_attr[10];
+nat_entry_t dbl_nat_entry;
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_NAT_TYPE;
+nat_entry_attr[0].value = SAI_NAT_TYPE_DOUBLE_NAT;
+
+nat_entry_attr[1].id = SAI_NAT_ENTRY_ATTR_SRC_IP;
+nat_entry_attr[1].value.u32 = 138.76.28.1; //example string
+
+nat_entry_attr[2].id = SAI_NAT_ENTRY_ATTR_DST_IP;
+nat_entry_attr[2].value.u32 = 200.200.200.100; //example string
+
+nat_entry_attr[3].id = SAI_NAT_ENTRY_ATTR_TO_ZONE;
+nat_entry_attr[3].value.u32 = 200;

```

Comment:
This attribute no longer exists, should be deleted

Reply:
removed